flow/transactions/sign/sign.go

```diff
@@ -0,0 +1,139 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sign
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/onflow/flow-go-sdk"
+	"github.com/psiemens/sconfig"
+	"github.com/spf13/cobra"
+
+	cli "github.com/onflow/flow-cli/flow"
+)
+
+type Config struct {
+	Signer                string   `default:"service" flag:"signer,s"`
+	Role                  string   `default:"authorizer" flag:"role"`
+	AdditionalAuthorizers []string `flag:"Additional authorizer addresses to add to the transaction"`
+	PayerAddress          string   `flag:"payer" info:"Specify payer of the transaction. Defaults to current signer."`
+	Code                  string   `flag:"code,c" info:"path to Cadence file"`
+	Host                  string   `flag:"host" info:"Flow Access API host address"`
+	Encoding              string   `default:"hexrlp" flag:"encoding" info:"Encoding to use for transactio (rlp)"`
+	Output                string   `default:"transaction.rlp" flag:"output,o" info:"Output location for transaction file"`
+}
+
+var conf Config
+
+var Cmd = &cobra.Command{
+	Use:   "sign",
+	Short: "Sign a transaction",
+	Run: func(cmd *cobra.Command, args []string) {
+		projectConf := cli.LoadConfig()
+
+		signerAccount := projectConf.Accounts[conf.Signer]
+		validateKeyPreReq(signerAccount)
+		var (
+			code  []byte
+			payer flow.Address
+			err   error
+		)
+
+		if conf.Code != "" {
+			code, err = ioutil.ReadFile(conf.Code)
+			if err != nil {
+				cli.Exitf(1, "Failed to read transaction script from %s", conf.Code)
+			}
+		}
+
+		if conf.PayerAddress != "" {
+			payer = flow.HexToAddress(conf.PayerAddress)
+		} else {
+			payer = signerAccount.Address
+		}
+
+		tx := flow.NewTransaction().
+			SetScript(code)
+
+		signerRole := cli.SignerRole(conf.Role)
+		switch signerRole {
+		case cli.SignerRoleAuthorizer:
+			tx.AddAuthorizer(signerAccount.Address)
+		case cli.SignerRolePayer:
+			if payer != signerAccount.Address {
+				cli.Exitf(1, "Role specified as Payer, but Payer address also provided, and different: %s !=", payer, signerAccount.Address)
+			}
+		case cli.SignerRoleProposer:
+			cli.Exitf(1, "Proposer role not yet supported: %s", conf.Role)
+		default:
+			cli.Exitf(1, "unknown role %s", conf.Role)
+		}
+
+		for _, authorizerString := range conf.AdditionalAuthorizers {
+			authorizerAddress := flow.HexToAddress(authorizerString)
+			tx.AddAuthorizer(authorizerAddress)
+		}
+
+		tx = cli.SignTransaction(projectConf.HostWithOverride(conf.Host), signerAccount, signerRole, tx, payer)
+
+		fmt.Printf("%s encoded transaction written to %s\n", conf.Encoding, conf.Output)
+
+		output := fmt.Sprintf("%x\n", tx.Encode())
+		err = ioutil.WriteFile(conf.Output, []byte(output), os.ModePerm)
```

Comment:
I think it can be a good idea to allow users to choose to write this to stdout as well as it will allow chaining of commands UNIX style. Maybe it would even be a good idea to do that as a default and if a user explicitly specifies the filename it saves it there, my thinking here is that command is called `sign` which implies signing operation more than saving - a similar example to `generate keys` command where you have `save` as a separate option. I think this could be a small change that would add to more consistency maybe.

Reply:
addressed in #110 which is now merged